src/thx/schema/SchemaDSLM.hx

```diff
@@ -0,0 +1,141 @@
+package thx.schema;
+
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.TypeTools;
+import thx.macro.MacroTypes;
+using thx.Arrays;
+
+class SchemaDSLM {
+  public static function makeVar(id: ExprOf<String>, rest: Array<Expr>) { //, constr: Expr, obj: ExprOf<Dynamic<Schema<Dynamic>>>) {
+    // trace(obj);
+    if(rest.length == 0)
+      Context.error('insufficient arguments', Context.currentPos());
+    if(rest.length > 3)
+      Context.error('too many arguments', Context.currentPos());
+    var constr = rest[0];
+    return if(rest.length == 1) {
+      macro thx.schema.SchemaDSL.constEnum($id, $constr);
+    } else if(rest.length == 2) {
+        var obj = rest[1];
+        var fields = SchemaDSLM.getFields(obj);
+        make(id, constr, value2Obj(constr, fields), obj);
+    } else {
+        var extr = rest[1];
+        var obj = rest[2];
+        make(id, constr, extr, obj);
+    }
+  }
+
+  public static function make(id: ExprOf<String>, constr: Expr, extr: Expr, obj: ExprOf<Dynamic<Schema<Dynamic>>>) {
```

Comment:
@fponticelli This is fascinating... Dynamic<T> is a type for some specific T? Is Dynamic a type constructor of an arbitrary number of arguments?